src/main/java/com/autotune/experimentManager/data/result/AutoTuneWorkFlow.java

```diff
@@ -0,0 +1,80 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package com.autotune.experimentManager.data.result;
+
+import com.autotune.experimentManager.handler.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.LinkedHashMap;
+
+/**
+ * Configuration class used to set workflow
+ * if do_experiments steps are prevalidation,deployment,postvalidation,load,metrics,summarizer,PostResults
+ * if do_monitoring only then prevalidation,load,metrics,summarizer,PostResults
+ * Load depends on up flag wait_for_load
+ */
+public class AutoTuneWorkFlow {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AutoTuneWorkFlow.class);
+    private final boolean wait_for_load;
+    private final LinkedHashMap<String, String> iterationWorkflowMap;
+    private final LinkedHashMap<String, String> trialWorkflowMap;
+
+    public AutoTuneWorkFlow(boolean do_experiments, boolean do_monitoring, boolean wait_for_load, String trialResultURL) throws Exception {
+        LOGGER.debug("Do_experiments : {} , Do_monitoring : {}", do_experiments, do_monitoring);
+        this.wait_for_load = wait_for_load;
+        iterationWorkflowMap = new LinkedHashMap<String, String>();
+        if (do_experiments == true) {
+            iterationWorkflowMap.put(PreValidationHandler.class.getSimpleName().replace("Handler", ""), PreValidationHandler.class.getName());
+            iterationWorkflowMap.put(DeploymentHandler.class.getSimpleName().replace("Handler", ""), DeploymentHandler.class.getName());
+            iterationWorkflowMap.put(PostValidationHandler.class.getSimpleName().replace("Handler", ""), PostValidationHandler.class.getName());
+            if (wait_for_load)
+                iterationWorkflowMap.put(LoadValidationHandler.class.getSimpleName().replace("Handler", ""), LoadValidationHandler.class.getName());
+            iterationWorkflowMap.put(MetricCollectionHandler.class.getSimpleName().replace("Handler", ""), MetricCollectionHandler.class.getName());
+            iterationWorkflowMap.put(SummarizerHandler.class.getSimpleName().replace("Handler", ""), SummarizerHandler.class.getName());
+        }
+        if (do_monitoring == true) {
+            iterationWorkflowMap.put(PreValidationHandler.class.getSimpleName().replace("Handler", ""), PreValidationHandler.class.getName());
+            if (wait_for_load)
+                iterationWorkflowMap.put(LoadValidationHandler.class.getSimpleName().replace("Handler", ""), LoadValidationHandler.class.getName());
+            iterationWorkflowMap.put(MetricCollectionHandler.class.getSimpleName().replace("Handler", ""), MetricCollectionHandler.class.getName());
+            iterationWorkflowMap.put(SummarizerHandler.class.getSimpleName().replace("Handler", ""), SummarizerHandler.class.getName());
+        }
```

Comment:
I think this is not right. In case we have both `do_experiments` and `do_monitoring` set to true, it will add `LoadValidationHandler` onwards twice

What we want instead is
```
        if (!do_experiments || !do_monitoring){
            throw new Exception("Workflow not defined");
        }
        if (do_experiments == true) {
            iterationWorkflowMap.put(PreValidationHandler.class.getSimpleName().replace("Handler", ""), PreValidationHandler.class.getName());
            iterationWorkflowMap.put(DeploymentHandler.class.getSimpleName().replace("Handler", ""), DeploymentHandler.class.getName());
            iterationWorkflowMap.put(PostValidationHandler.class.getSimpleName().replace("Handler", ""), PostValidationHandler.class.getName());
}
        if (wait_for_load) {
                iterationWorkflowMap.put(LoadValidationHandler.class.getSimpleName().replace("Handler", ""), LoadValidationHandler.class.getName());
        }
        if (do_monitoring == true) {
            iterationWorkflowMap.put(MetricCollectionHandler.class.getSimpleName().replace("Handler", ""), MetricCollectionHandler.class.getName());
            iterationWorkflowMap.put(SummarizerHandler.class.getSimpleName().replace("Handler", ""), SummarizerHandler.class.getName());
        }
```

Reply:
Thanks Dino .Fixed now.